aten/src/ATen/native/cuda/UnaryOpsKernel.cu

```diff
@@ -71,10 +71,48 @@ void erfinv_kernel_cuda(TensorIterator& iter) {
   });
 }
 
+void digamma_kernel_cuda(TensorIterator& iter) {
+  if (iter.dtype() == kHalf) {
+    gpu_kernel(iter, []GPU_LAMBDA(at::Half a) -> at::Half {
+        return calc_digamma<float>(a);
+    });
+  } else {
+    AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "digamma_cuda", [&]() {
+      gpu_kernel(iter, []GPU_LAMBDA(scalar_t a) -> scalar_t {
+        return calc_digamma(a);
+      });
+    });
+  }
+}
+
+void trigamma_kernel_cuda(TensorIterator& iter) {
+  if (iter.dtype() == kHalf) {
+    gpu_kernel(iter, []GPU_LAMBDA(at::Half a) -> at::Half {
+        return calc_trigamma<float>(a);
+    });
+  } else {
+    AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "trigamma_cuda", [&]() {
+      gpu_kernel(iter, []GPU_LAMBDA(scalar_t a) -> scalar_t {
+        return calc_trigamma(a);
+      });
+    });
+  }
+}
+
+void polygamma_kernel_cuda(TensorIterator& iter, int64_t n) {
+  switch (n) {
+    case 0: digamma_kernel_cuda(iter); break;
+    case 1: trigamma_kernel_cuda(iter); break;
+    default: AT_ERROR("polygamma(n,x) is not implemented for n>=2");
```

Comment:
If the error message is shown here, I believe it should be:

    AT_ERROR("polygamma(n,x) is not implemented for n>=2 on CUDA");

An error handling for `n<0` is also missing.

Reply:
We should also use TORCH_CHECK instead of AT_ERROR; AT_ERROR is deprecated

Reply:
changed